x-pack/plugins/index_management/public/application/sections/home/index_list/index_table/index_table.js

```diff
@@ -65,17 +66,26 @@ const getColumnConfigs = ({
         defaultMessage: 'Name',
       }),
       order: 10,
-      render: (index) => (
-        <>
-          <EuiLink
-            data-test-subj="indexTableIndexNameLink"
-            onClick={() => history.push(getIndexDetailsLink(index.name, location.search || ''))}
-          >
-            {index.name}
-          </EuiLink>
-          {renderBadges(index, extensionsService, filterChanged)}
-        </>
-      ),
+      render: (index) => {
+        return (
+          <>
+            <EuiLink
+              data-test-subj="indexTableIndexNameLink"
+              onClick={() => {
+                if (!extensionsService.indexDetailsPageRoute) {
+                  history.push(getIndexDetailsLink(index.name, location.search || ''));
+                } else {
+                  const route = extensionsService.indexDetailsPageRoute.renderRoute(index.name);
+                  application.navigateToUrl(route);
```

Comment:
This will work in serverless but in stateful I think you need to prepend the base path.

`application.navigateToUrl(http.basePath.prepend(route));`

we could do the `http.basePath.prepend(route)` in the `renderRoute` function too, we have `coreStart` available there.

Reply:
> This will work in serverless but in stateful I think you need to prepend the base path.

Curious, why this wouldn't work in stateful ?

Reply:
because it stateful the URL has a prefix, and sometimes has a space prefix, add the path as well. So you need to prepend those to any app URL you are navigating to directly.

Reply:
Makes sense thank you for explaining.

Reply:
added in 6bdd523f772c3dfe7b46f4ce81811bc4123e6f1b